### DISCUSSION TITLE
react-admin data provider for FastAPI

### DISCUSSION THREAD

**Discussion by tech4242** [fastapi/fastapi#8893] [Questions]
Some of you might know `react-admin` i.e. https://marmelab.com/react-admin/ it's a pretty popular React admin UI that can consume REST APIs in an easy way through what they call "Data Providers" which is nothing more than a nice-ish boilerplate to consume, parse and display API responses.

If you want to use their "standard" json provider you get hit with the following msg:

_Error: The X-Total-Count header is missing in the HTTP Response. The jsonServer Data Provider expects responses for lists of resources to contain this header with the total number of results to build the pagination. If you are using CORS, did you declare X-Total-Count in the Access-Control-Expose-Headers header?_

There is also a pretty cool provider for Django REST Framework: https://github.com/synaptic-cl/ra-data-drf

And I was wondering if:
1) Anyone is using react-admin with FastAPI 
and if yes 
2) Which provider did you decide to go for and did you end up manually setting headers like `X-Total-Count`

Probably a good OSS project to work on unless someone has already started or is using one of the other providers: https://marmelab.com/react-admin/DataProviders.html#available-providers

Thank you in advance.

P.S. I thought I would get more relevant responses here vs. the react-admin repo since everyone here is building FastAPI projects and the react-admin project is popular enough so I am sure I am not the first person thinking about this :)

**Comment by achaussier** [comment]
Hello,

I think about Fast API for a new project and have the same question (but one year later ;)) to evaluate the work to plan.

Are some of you using react-admin with FastApi (or perhaps another admin project should be preferred ) ?

Thanks for your responses and have a nice day
